sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -205,6 +207,13 @@ public String getBaseOutputFilename() {
     return baseOutputFilename.get();
   }
 
+  /**
```

Comment:
`getBaseOutputFilename` and `getBaseOutputFilenameProvider` -- is it reasonable to merge?